e2e/tests/upgrades/upgrade_test.go

```diff
@@ -405,3 +551,29 @@ func getICAVersion(chainAVersion, chainBVersion string) string {
 	// explicitly set the version string because the host chain might not yet support incentivized channels.
 	return icatypes.NewDefaultMetadataString(ibctesting.FirstConnectionID, ibctesting.FirstConnectionID)
 }
+
+// Status queries the status of the client by clientID
+func (s *UpgradeTestSuite) Status(ctx context.Context, chain ibc.Chain, clientID string) (string, error) {
+	queryClient := s.GetChainGRCPClients(chain).ClientQueryClient
+	res, err := queryClient.ClientStatus(ctx, &clienttypes.QueryClientStatusRequest{
+		ClientId: clientID,
+	})
+	if err != nil {
+		return "", err
+	}
+
+	return res.Status, nil
+}
+
+// ClientState queries the current ClientState by clientID
+func (s *UpgradeTestSuite) ClientState(ctx context.Context, chain ibc.Chain, clientID string) (*clienttypes.QueryClientStateResponse, error) {
```

Comment:
I believe this function exists in `testsuite/grpc_query.go`

Reply:
so I am leaving this one, bc the function in grpc query unpacks the client state there which runs into the issue that the ClientState cannot be Unpacked bc it is trying to unpack it from a legacy type which is unsupported